backend_new/src/services/application-csv-export.service.ts

```diff
@@ -0,0 +1,596 @@
+import fs from 'fs';
+import path from 'path';
+import { Inject, Injectable, Scope } from '@nestjs/common';
+import { REQUEST } from '@nestjs/core';
+import { Request as ExpressRequest } from 'express';
+import { view } from './application.service';
+import { PrismaService } from './prisma.service';
+import { MultiselectQuestionService } from './multiselect-question.service';
+import { ApplicationCsvQueryParams } from '../dtos/applications/application-csv-query-params.dto';
+import { SuccessDTO } from '../dtos/shared/success.dto';
+import { UnitType } from '../dtos/unit-types/unit-type.dto';
+import { Address } from '../dtos/addresses/address.dto';
+import { ApplicationMultiselectQuestion } from 'src/dtos/applications/application-multiselect-question.dto';
+import MultiselectQuestion from '../dtos/multiselect-questions/multiselect-question.dto';
+import { ApplicationFlaggedSet } from '../dtos/application-flagged-sets/application-flagged-set.dto';
+import { UnitTypeService } from './unit-type.service';
+
+view.csv = {
+  ...view.details,
+  applicationFlaggedSet: true,
+  listings: false,
+};
+
+type CsvHeader = {
+  path: string;
+  label: string;
+  format?: (val: unknown) => unknown;
+};
+
+@Injectable({ scope: Scope.REQUEST })
```

Comment:
instead of injecting scope, instead passing the req object from the controller down into the service's function means the service won't get re-instantiated each request

Reply:
removed the injected scope